Add deprecation notice Fixes #50 (#52)
pytest-dev/pytest-runner: README.rst

Setup scripts can use pytest-runner to add setup.py test support for pytest
runner.

Deprecation Notice
==================

> pytest-runner depends on deprecated features of setuptools and relys on features that break security

Author: ```suggestion
pytest-runner depends on deprecated features of setuptools and relies on features that break security
```

> mechanisms in pip. For example `setup_requires` and `tests_require` bypass `pip --require-hashes`.

Author: ```suggestion
mechanisms in pip. For example 'setup_requires' and 'tests_require' bypass ``pip --require-hashes``.
```

See also https://github.com/pypa/setuptools/issues/1684.

It is recommended that you:

- Remove 'pytest-runner' from your 'setup_requires', preferably removing the `setup_requires` option.
- Remove 'pytest' and any other testing requirements from 'tests_require', preferably removing the `setup_requires` option.

> - Use a requirement-test.txt file to hold test dependencies

Reviewer: I don't think there is such a section yet. I wish there was.

> - Invoke tests with ``pytest``

Author: ```suggestion
- Invoke tests with ``pip install --require-hashes requirements-test.txt && pytest``
```

> - Invoke tests with ``pip install --require-hashes requirements-test.txt && pytest``

Reviewer: This syntax is hostile to Windows and also mutates the user's environment. Especially for a user coming from `pytest-runner`, which doesn't mutate the environment, that might be a surprise.  If a one-liner is required, I'd suggest instead `pip-run -r tests/requirements.txt -- -m pytest`. Of course, that assumes pip-run and not just pip. I think better would be to recommend a generic "select a tool to bootstrap and then run tests such as tox". Maybe link to tox or link to a section in Python packaging docs that may enumerate options and/or give the (current) best recommendation.

- Another good option is to configure a ``tox.ini`` next to your ``setup.cfg`` https://tox.readthedocs.io/en/latest/

> Usage

Author: Do you think it's worth removing this part of the docs, in case people scroll past the deprecation warning?

Reviewer: As long as the package exists, I think having docs is worthwhile.
